### ISSUE TITLE
Accessibility Strategy doesn't work in Firefox

### ISSUE THREAD

**Issue by godbout** [godbout/kindaVim.theapp#70]
1. first investigation shows that Firefox returns that the AX is enabled while it's not true, so kV cannot enforce special attributes like it does for Electron apps and Chromium browsers
2. once the AX is enforced, changing/deleting doesn't work properly as it doesn't seem to act on the right portion of the selection. not sure yet whether it's the AX that is too slow in Firefox (like in Chromium and Electron), or if Firefox just plainly returns wrong data

to investigate more.

**Comment by godbout** [comment]
ok so after investigating, there's two issues. one controllable (by me), one not:

1. when i ping Firefox/LibreWolf, it tells me the AX is enabled, while it's not. so i can't use the same system as with Electron/Chromium. i need to basically test that the current frontmost app is specifically Firefox/LibreWolf, and set the AX on at each keystroke. it's quite cumbersome, and inefficient, but it works. Firefox's AX can be set totally programmatically, for LibreWolf it can't. it's locked by default, and needs to be enabled in LibreWolf settings (see video).
2. reading with the AX can work, but writing will not. it doesn't handle caret location and selection length correctly (see video). Mozilla on their own site claims that they handle "basic" VoiceOver support: https://support.mozilla.org/en-US/kb/accessibility-features-firefox-make-firefox-and-we#w_screen-reader. i guess that's what they mean by basic. no writing. so we would only be able to "move", like `w`, `j` etc. but we wouldn't  be able to change/delete the text, like `cc`, `daw`, etc. so there's no point in allowing the Accessibility Strategy for now in Firefox/LibreWolf. for this part we need to ask the Mozilla team to fix/work on this (i know they're having less people working on Firefox now, so no expecting this to happen).

https://user-images.githubusercontent.com/121373/168432514-0be9346f-8410-4441-b024-3a7f218f3239.mp4

if i implement point 1, that means we could still have Firefox/LibreWolf working in the `Key Mapping` Family. currently it would only work in the `911` one.

the difference between the `Key Mapping` and the `911` is that the `Key Mapping` handles `Text Elements` and `Non Text Elements` differently, while the `911` considers everything as a `Text Element`.
for example for the `Key Mapping`, for a `Text Element`, kV translates `dd` by `control a, control shift e, shift right, command c, delete` while for a `Non Text Element` (like an item in a list) it simply translates it by `delete`.
the `911` would always translate using the `Text Element` translation.

so i guess the question now, how was the experience using the `911` in Firefox/LibreWolf so far? is it workable?

**Comment by mrjones2014** [comment]
It's pretty workable leaving Firefox/Librewolf in 911 family. The only thing that doesn't work (and isn't expected to work) is the `gx` mapping, but that's not a huge deal. 

It's a real shame to see Firefox basically dying. Chrome having such dominance is pretty scary.

**Comment by godbout** [comment]
> It's pretty workable leaving Firefox/Librewolf in 911 family. The only thing that doesn't work (and isn't expected to work) is the `gx` mapping, but that's not a huge deal.

alright. i'll keep the issue open for now. when i'll work on the wizard i'll have to take care of each app individually. so maybe i'll find a way to get Firefox/LibreWolf work with the `Key Mapping`. would be a bit better.

> 
> It's a real shame to see Firefox basically dying. Chrome having such dominance is pretty scary.

wasn't there some changes in their management or plans some years ago at Mozilla? tbh i mostly use computers to build stuff, and i decided to stick to Macs when i started kV. i use Safari for my day-to-day stuff and i like it. even more with all the native AX, many stuff possible. switch to Brave (made an Alfred Workflow that sends the current URL from Safari to Brave and vice-versa) when i web dev coz i'm used to the dev tools, or when i want to watch a video double the speed (Safari has some metallic artifacts when you speed up sound). Chrome i stay away, was always killing my laptop and spawning a bunch of shits in the background. urgh. but each time i develop something, i make sure it works well in Safari coz that's what i use, then in a Chromium base browser coz as you said, the dominance is insane. never try in Firefox :p my bad.

ok and about the other issue...ah forget it, let's not talk about it.

**Comment by godbout** [comment]
bug reported: https://bugzilla.mozilla.org/show_bug.cgi?id=1769412

**Comment by mrjones2014** [comment]
I avoid Chrome and any Chromium based browsers as a rule, for a lot of reasons. Brave specifically has done some very icky things like silently replacing crypto links such as Binance with their own affiliate links, profiting from it without user knowledge or consent. Also Brave Software is an ad company. Overall I think Brave is  a red herring and I usually recommend people against it who are really serious about privacy (privacy is the big thing I’m passionate about).

this guy sums up really well why chrome dominance spells doom for the web: https://youtu.be/eA8O97U1Pbc

**Comment by godbout** [comment]
isn't the guy from Brave one of the creator of JS and Mozilla? anyways i'm a simple guy, i can't deal with many steps/priorities, etc. at the same time. i usually focus on 3 criteria max. for software, privacy is not one of them (as a user). i even allow tracking if i happen to use a free app. but i'm getting to dislike Brave for another reason. i switched from Chrome because there was too much shits happening in the background, sucking my computer etc. now, Brave sends 3 thousand popups, ads here and there, notifications, blah blah blah. i want to get the job done straight. i don't seem to be able to anymore. reminds me something that seems to get trendy and i absolutely can't understand: i like to support and buy apps from small indie developer(s), but many of them, even after i bought their apps, show me some fucking alert to rate and/or write a review for their app. like right when i'm using it. an app i bought to get shits done and remove friction. i can't even rationalize how this is for the user rather than for the developer. baffles me. what a sad world. here let me get another glass of whisky.

**Comment by godbout** [comment]
not sure i make much sense when i'm digressing like this. i'm trying to trim the fat coz else i would write a PhD thesis.

**Comment by godbout** [comment]
the Mozilla team asked for more info. hopefully they'll follow up: https://bugs.chromium.org/p/chromium/issues/detail?id=1325643#c5

**Comment by godbout** [comment]
@mrjones2014 maybe you want to be careful with Firefox... see: https://www.xn--80ak6aa92e.com/ :D

**Comment by godbout** [comment]
> @mrjones2014 maybe you want to be careful with Firefox... see: https://www.xn--80ak6aa92e.com/ :D

LibreWolf patched against this.

**Comment by mrjones2014** [comment]
> @mrjones2014 maybe you want to be careful with Firefox... see: https://www.xn--80ak6aa92e.com/ :D

What's the URL? It's too sus for me to click it 😆 also it's in one of the blocklists on my Pi-Hole lol

**Comment by godbout** [comment]
![Screen Shot 2022-05-26 at 00 33 17](https://user-images.githubusercontent.com/121373/170314999-1f530e62-0687-4686-b039-257140a55b06.png)

![Screen Shot 2022-05-26 at 00 33 37](https://user-images.githubusercontent.com/121373/170315019-09fd9a14-b4bf-4955-8ff6-29abd8e32bbf.png)

the URL is using Unicode encoding to pretend it's a site it's not (like `a` is not an ANSI `a`, it's a Cyrillic one and looks the same).

**Comment by godbout** [comment]
i'm not gonna send a link about DuckDuckGo allowing Microsoft trackers because of their search arrangement. enough for today.

**Comment by mrjones2014** [comment]
lol, yeah I already saw that. stuff like that is why I block all JS by default. JS is a privilege, and I grant it very sparingly 😆

**Comment by godbout** [comment]
sorry. just recording a video for Wooshy.

**Comment by godbout** [comment]
sorry. another try at the video.

**Comment by godbout** [comment]
that was the video btw: https://www.youtube.com/watch?v=-N6VwfQGrCU

**Comment by godbout** [comment]
another one. sorry.

**Comment by godbout** [comment]
still confirmed that Firefox/Librewolf is Key Mapping coz Mozilla still hasn't done shit. all in The Wizard. will be updated when available. closing.
